operation/manager.go

```diff
@@ -137,6 +154,19 @@ func (sm *SingleOperationManager) cancelInLock(ctx context.Context) {
 	sm.currentOp = nil
 }
 
+func (sm *SingleOperationManager) cancelNew(ctx context.Context) {
```

Comment:
I don't think this is right - though maybe all we need is to remove the waitCh entirely instead of making a new function. looking at https://github.com/viamrobotics/rdk/pull/766/files#diff-b5dfc1ffe50cc40f2798c5285b1d3fb7f1a7ec8f86d8ddee9fb5ebf2baefa27dL263,
the purpose of the channel close is to make sure that we've finished the cleanup. However, in this case, the cleanup is literally just setting currentOp to nil, which we already do in both the cleanup func and the cancelInLock func. So waitCh might not be necessary?

Reply:
Yupp, I agree. I removed the wait on waitCh in cancelInLock and removed the new function as well